test-app/tests/integration/components/bs-modal/header-test.gts

```diff
@@ -88,24 +97,34 @@ module('Integration | Component | bs-modal/header', function (hooks) {
 
   test('close button can be removed', async function (assert) {
     await render(
-      hbs`<BsModal::Header @title='Header' @closeButton={{false}} />`,
+      <template>
+        <BsModalHeader @title='Header' @closeButton={{false}} />
+      </template>,
     );
 
     assert
       .dom(`.modal-header button.${closeButtonClass()}`)
       .doesNotExist('Modal header has no close button.');
   });
 
-  test('close button can be removed in yield block form', async function (assert) {
-    await render(hbs`{{! template-lint-disable no-unused-block-params }}
-<BsModal::Header as |header|><div id='custom'>Test</div></BsModal::Header>
-{{! template-lint-enable no-unused-block-params }}`);
+  test('close button can be removed in yield block form', async function (this: RenderingTestContext, assert) {
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    await render(
+      <template>
+        {{! template-lint-disable no-unused-block-params }}
+        <BsModalHeader as |header|>
+          <div id='custom'>Test</div>
+        </BsModalHeader>
+        {{! template-lint-enable no-unused-block-params }}
+      </template>,
+    );
+    /* eslint-enable @typescript-eslint/no-unused-vars */
```

Comment:
Why can we just not use the yielded value?

```suggestion
    await render(
      <template>
        <BsModalHeader>
          <div id='custom'>Test</div>
        </BsModalHeader>
      </template>,
    );
```

Reply:
I think this is testing the has-block-params codepath, so this might have been intentional in the original test. Converting it to GTS just makes the linter more strict

Reply:
If this is about testing the block params we should also use them. Otherwise compiler may refactor them away as they are not used and we don't notice it.

Reply:
That makes sense. I've updated the test to use the title out of the block params